### ISSUE TITLE
Create filtering endpoint

### ISSUE THREAD

**Issue by LeonByrne** [mortgage-analytics/backend#43]
We need a filtering endpoint, and required methods in services and repos, for the stage transition page. We will be filtering applications based on the date it was created (get between two dates), application type, whether it is single or joint, application stage, application status, etc (the story ticket lists some others but ignore them, I'm not sure if they're still present in the new data). The endpoint should return all applications that fit these filters.

To recap filters are
Started between Dates
Application type
whether it is single or joint
application stage
application status
